### PR TITLE
RecordParser: Limit the record size

### PR DESCRIPTION
**Pull request by richardf**
A simple implementation of  #2560.

### CODE REVIEW

File: src/main/java/io/vertx/core/parsetools/RecordTooLongException.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.core.parsetools;
+public class RecordTooLongException extends RuntimeException {
```
Comment (vietj) [inline src/main/java/io/vertx/core/parsetools/RecordTooLongException.java]: can you just use a `VertxException` instead ?

Comment (vietj) [reply src/main/java/io/vertx/core/parsetools/RecordTooLongException.java]: or better an `IllegalStateException`

Comment (richardf) [reply src/main/java/io/vertx/core/parsetools/RecordTooLongException.java]: Sure

**Comment by vietj** [comment]
back-port to 3.5.4

**Pull request merged**